src/datasources/asset-calibration/AssetCalibrationDataSource.ts

```diff
@@ -45,18 +45,40 @@ export class AssetCalibrationDataSource extends DataSourceBase<AssetCalibrationQ
 
     result.fields = calibrationForecastResponse.calibrationForecast.columns || [];
     result.fields = result.fields.map(field => this.formatField(field, query));
+    result.fields = this.formatResultsWhenNoTimeSpecified(result, query);
```

Comment:
This logic is kind of odd to me. Line 47 does nothing if it's a location/model grouping, and line 48 does nothing if it's a time-based grouping. I think there should be two different "pipelines" depending on which response format we get, something like:

```js
if (isGroupedByTime(query)) {
  this.processResultsGroupedByTime(...)
} else {
  this.processResultsGroupedByCategory(...)
}

Reply:
@mure I agree with you, we can write it in a more intuitive way.